src/NuGet.Clients/NuGet.PackageManagement.UI/Converters/IconUrlToImageCacheConverter.cs

```diff
@@ -86,49 +77,52 @@ public object Convert(object[] values, Type targetType, object parameter, Cultur
-                        var par = lazyPar.Value as PackageArchiveReader;
-                        var iconEntry = Uri.UnescapeDataString(iconUrl.Fragment).Substring(1); // skip the '#' in a URI fragment
-                        iconBitmapImage.StreamSource = par.GetEntry(iconEntry).Open(); // This stream is closed in BitmapImage events
-
-
-                        iconBitmapImage.DecodeFailed += (sender, args) =>
+                        var reader = lazyPar(); // Always returns a new reader. That avoids using an already disposed one
+                        if (reader is PackageArchiveReader par) // This reader is closed in BitmapImage events
                         {
-                            par.Dispose();
-                            IconBitmapImage_DownloadOrDecodeFailed(sender, args);
-                            AddToCache(iconUrl, defaultPackageIcon);
-                        };
-                        iconBitmapImage.DownloadFailed += (sender, args) =>
+                            iconBitmapImage.StreamSource = par.GetEntry(iconEntry).Open();
+
+                            iconBitmapImage.DecodeFailed += (sender, args) =>
+                            {
+                                reader.Dispose();
+                                IconBitmapImage_DownloadOrDecodeFailed(sender, args);
+                                AddToCache(iconUrl, defaultPackageIcon);
+                            };
+                            iconBitmapImage.DownloadFailed += (sender, args) =>
+                            {
+                                reader.Dispose();
+                                IconBitmapImage_DownloadOrDecodeFailed(sender, args);
+                                AddToCache(iconUrl, defaultPackageIcon);
+                            };
+                            iconBitmapImage.DownloadCompleted += (sender, args) =>
+                            {
+                                reader.Dispose();
+                                IconBitmapImage_DownloadCompleted(sender, args);
+                            };
+
+                            imageResult = FinishImageProcessing(iconBitmapImage, iconUrl, defaultPackageIcon);
+                        }
+                        else // we cannot use the reader object
                         {
-                            par.Dispose();
-                            IconBitmapImage_DownloadOrDecodeFailed(sender, args);
                             AddToCache(iconUrl, defaultPackageIcon);
-                        };
-                        iconBitmapImage.DownloadCompleted += (sender, args) =>
-                        {
-                            par.Dispose();
-                            IconBitmapImage_DownloadCompleted(sender, args);
-                        };
-
-                        imageResult = FinishImageProcessing(iconBitmapImage, iconUrl, defaultPackageIcon);
+                            imageResult = defaultPackageIcon;
```

Comment:
Also in this `else` block:
```C#
reader?.Dispose();
```

Reply:
Added the reader disposal at the beginning of the else block